charts/tidb-cluster/templates/scripts/_start_drainer.sh.tpl

```diff
@@ -27,7 +27,7 @@ done
 /drainer \
 -L={{ .Values.binlog.drainer.logLevel | default "info" }} \
 -pd-urls={{ template "cluster.scheme" . }}://{{ template "cluster.name" . }}-pd:2379 \
--addr=`echo ${HOSTNAME}`.{{ template "cluster.name" . }}-drainer:8249 \
+-addr=:8249 \
```

Comment:
able to port forward  to 8249 (k8s [hard coded host as localhost](https://github.com/kubernetes/kubernetes/blob/928817a26a84d9e3076d110ea30ba994912aa477/pkg/kubelet/dockershim/docker_streaming_others.go#L47))

Reply:
This might cause the existing drainer to be restarted. I'm not sure if we should allow this for major tidb-operator upgrade.

Reply:
I'll revert this. Alternatively, we can exec into the pod to check the API.

Reply:
We can make this configurable and use the original one by default, in e2e, switch to listen on `0.0.0.0`

Reply:
yes, it's a good solution. But I'm still going to use `exec` API because this is a general solution to access k8s resources too. K8S e2e framework provides [a lot of utilities] (https://github.com/kubernetes/kubernetes/blob/e76619931f805e3ba2ceece405b662891810c5e7/test/e2e/framework/exec_util.go#L50) around `exec` API. It's simple to use. No more dependencies and no need to change current charts.

Port forwarding is more useful in MySQL query checks because images normally do not contain MySQL client.

Reply:
implemented: https://github.com/pingcap/tidb-operator/pull/1304/files#diff-67dacabf080ca4dc8e95d2acf12ee36bR3161-R3171 
the alpine image has builtin wget (busybox)